sdks/wasm/src/library_mono.js

```diff
@@ -141,6 +141,27 @@ var MonoSupportLib = {
 			return final_var_list;
 		},
 
+		// code from https://stackoverflow.com/a/5582308
+		//
+		// Given `dotnet:object:foo:bar`,
+		// returns [ 'dotnet', 'object', 'foo:bar']
+		_split_object_id: function (id, delim = ':', count = 3) {
```

Comment:
Would it make more sense to think of this as parse and return an object with symbolic names rather than positional?

Reply:
I changed this in the valuetypes PR - https://github.com/mono/mono/pull/20045/commits/85ff76298061689e9d70f30a4155af0ee37c3c53 .